libtransmission/torrent-files.cc

```diff
@@ -310,3 +310,114 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
         depthFirstWalk(filename.c_str(), remove_junk);
     }
 }
+
+namespace
+{
+
+// https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+// Do not use the following reserved names for the name of a file:
+// CON, PRN, AUX, NUL, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8,
+// COM9, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9.
+// Also avoid these names followed immediately by an extension;
+// for example, NUL.txt is not recommended.
+[[nodiscard]] bool isReservedFile(std::string_view in) noexcept
+{
+    if (std::empty(in))
+    {
+        return false;
+    }
+
+    // Shortcut to avoid extra work below.
+    // All the paths below involve filenames that begin with one of these chars
+    static auto constexpr ReservedFilesBeginWithOneOf = "ACLNP"sv;
+    if (ReservedFilesBeginWithOneOf.find(toupper(in.front())) == std::string_view::npos)
+    {
+        return false;
+    }
+
+    auto in_upper = tr_pathbuf{ in };
+    std::transform(std::begin(in_upper), std::end(in_upper), std::begin(in_upper), [](auto ch) { return toupper(ch); });
+    auto const in_upper_sv = in_upper.sv();
+
+    static auto constexpr ReservedNames = std::array<std::string_view, 22>{
+        "AUX"sv,  "CON"sv,  "NUL"sv,  "PRN"sv,  "COM1"sv, "COM2"sv, "COM3"sv, "COM4"sv, "COM5"sv, "COM6"sv, "COM7"sv,
+        "COM8"sv, "COM9"sv, "LPT1"sv, "LPT2"sv, "LPT3"sv, "LPT4"sv, "LPT5"sv, "LPT6"sv, "LPT7"sv, "LPT8"sv, "LPT9"sv,
+    };
+    if (std::find(std::begin(ReservedNames), std::end(ReservedNames), in_upper_sv) != std::end(ReservedNames))
+    {
+        return true;
+    }
+
+    static auto constexpr ReservedPrefixes = std::array<std::string_view, 22>{
+        "AUX."sv, "CON."sv,  "NUL."sv,  "PRN."sv,  "COM1."sv, "COM2"sv,  "COM3."sv, "COM4."sv, "COM5."sv, "COM6."sv, "COM7."sv,
+        "COM8"sv, "COM9."sv, "LPT1."sv, "LPT2."sv, "LPT3."sv, "LPT4."sv, "LPT5."sv, "LPT6."sv, "LPT7."sv, "LPT8."sv, "LPT9."sv,
```

Comment:
```suggestion
        "AUX."sv, "CON."sv,  "NUL."sv,  "PRN."sv,  "COM1."sv, "COM2."sv,  "COM3."sv, "COM4."sv, "COM5."sv, "COM6."sv, "COM7."sv,
        "COM8."sv, "COM9."sv, "LPT1."sv, "LPT2."sv, "LPT3."sv, "LPT4."sv, "LPT5."sv, "LPT6."sv, "LPT7."sv, "LPT8."sv, "LPT9."sv,
```

Randomly reading PRs and following along and saw this after seeing the other comment. No idea what the code does.

Reply:
Ah... I see. Good random fact to know :)